tests/code/test_event_list.py

```diff
@@ -0,0 +1,79 @@
+        studentProgramEvent = ProgramEvent.create(program = 2, event = Studentled)
+
+        training = Event.create(name = "Test Training Program",
+                                term = 3,
+                                description = "event for testing",
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "basement",
+                                isTraining = 1,
+                                startDate = 2021-12-12,
+                                endDate = 2021-12-13)
+        trainingProgramEvent = ProgramEvent.create(program = 2, event = training)
+
+        bonner = Event.create(name = "Test Bonner Program",
+                                term = 3,
+                                description = "event for testing",
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "basement",
+                                startDate = 2021-12-12,
+                                endDate = 2021-12-13)
+        bonnerProgramEvent = ProgramEvent.create(program = 5, event = bonner)
+
+
+        oneTime = Event.create(name = "Test One Time",
+                                term = 3,
+                                description = "event for testing",
+                                timeStart = "18:00:00",
+                                timeEnd = "21:00:00",
+                                location = "basement",
+                                startDate = 2021-12-12,
+                                endDate = 2021-12-13)
+        oneTimeProgramEvent = ProgramEvent.create(program = 6, event = oneTime)
+
+        studentledProgram = getStudentLedProgram(3)
+        trainingProgram = getTrainingProgram(3)
+        trainingProgram2 = getTrainingProgram(2)
+        bonnerProgram = getBonnerProgram(3)
+        oneTimeEvents = getOneTimeEvents(3)
+
+        assert studentledProgram
+        studentledRes = []
+        for program, events in studentledProgram.items():
+            for event in events:
+                studentledRes.append(event.name)
+        assert "Test Student Lead" in studentledRes
+
+        assert trainingProgram
+        assert True in [event.name == "Test Training Program" for event in trainingProgram]
+        assert True not in [event.name == "Test Student Lead" for event in trainingProgram]
+        assert True not in [event.name == "Test Training Program" for event in trainingProgram2]
+
+        assert bonnerProgram
+        assert True in [event.name == "Test Bonner Program" for event in bonnerProgram]
+        assert True not in [event.name == "Test Student Lead" for event in bonnerProgram]
+
+        assert oneTimeEvents
+        assert True in [event.name == "Test One Time" for event in oneTimeEvents]
+        assert True not in [event.name == "Test Student Lead" for event in oneTimeEvents]

```

Comment:
For these asserts, you have an event object you can use to directly check for inclusion (or not) in the event list.